EssentialsGeoIP/src/com/earth2me/essentials/geoip/EssentialsGeoIPPlayerListener.java

```diff
@@ -175,6 +175,12 @@ private void downloadDatabase() {
                 logger.log(Level.SEVERE, tl("geoIpUrlEmpty"));
                 return;
             }
+            String licenseKey = config.getString("database.license-key", "");
+            if (licenseKey == null || licenseKey.isEmpty()) {
+                logger.log(Level.SEVERE, tl("geoIpLicenseMissing"));
+                return;
+            }
+            url = url.replace("{LICENSEKEY}", licenseKey);
```

Comment:
Consider replace all maybe? idk.

Reply:
The license key only appears in the URL once. I think it would be very strange if it appeared more than that.

Reply:
You could probably put the license key at the end of the URL if that was easier.  Then just append the value from the config.

Reply:
That is ironically the exact reason why I was having troubles before. I had the suffix query parameter in the middle rather than at the end, which caused `url.endsWith(".gz")` to break down. I think it's better to keep it a placeholder (in the same fashion as the base EssentialsX module does in its config). There technically isn't any requirement for the license key to be at the end, so I don't think it would be wise to impose that on users (especially if they use a custom download link).